GraphResponse.md

```diff
@@ -7,6 +7,206 @@ Provide a decorator for the native response message to enable attaching Graph sp
+
+## The GraphResponse Object
+
+The GraphResponse object can wrap the native HttpResponseMessage object so that customers can use the familiar HttpResponseMessage members. 
+GraphResponse will be part of Microsoft.Graph.Core and it will expose the headers,Content and status code while also keeping reference to the Request object as below.
+
+```csharp
+
+    /// <summary>
+    /// The GraphResponse Object
+    /// </summary>
+    public class GraphResponse 
+    {
+        /// <summary>
+        /// The GraphResponse Constructor
+        /// </summary>
+        /// <param name="iBaseRequest">The Request made for the response</param>
+        /// <param name="httpResponseMessage">The response</param>
+        public GraphResponse(IBaseRequest iBaseRequest,HttpResponseMessage httpResponseMessage)
+        {
+            this.httpResponseMessage = httpResponseMessage;
+            this.BaseRequest = iBaseRequest;
+        }
+
+        private readonly HttpResponseMessage httpResponseMessage;
+        
+        /// <summary>
+        /// The Response Status code
+        /// </summary>
+        public HttpStatusCode StatusCode => httpResponseMessage.StatusCode;
+        
+        /// <summary>
+        /// The Response Content
+        /// </summary>
+        public HttpContent Content => httpResponseMessage.Content;
+        
+        /// <summary>
+        /// The Response Headers
+        /// </summary>
+        public HttpResponseHeaders HttpHeaders => httpResponseMessage.Headers;
+
+        /// <summary>
+        /// The reference to the Request
+        /// </summary>
+        public IBaseRequest BaseRequest;
+
+        /// <summary>
+        /// Get the native Response Message
+        /// </summary>
+        /// <returns></returns>
+        public HttpResponseMessage ToHttpResponseMessage()
+        {
+            return httpResponseMessage;
+        }
+    }
+
+```
+
+## Exposing the GraphResponse object from the Service Library
+
+We can create parallel *Async functions to enable the GraphResponse object functionality. We will generate requests with the following public API signatures:
+
+* `GetWithGraphResponseAsync(): : GraphResponse`
+* `CreateWithGraphResponseAsync(NewObject: Entity) : GraphResponse`
+* `PostWithGraphResponseAsync(NewObject: Entity) : GraphResponse`
+* `UpdateWithGraphResponseAsync(UpdatedObject: Entity) : GraphResponse`
+* `DeleteWithGraphResponseAsync() : GraphResponse` 
+
+These would in turn call a function in the core library(BaseRequest) that looks something like
+
+```csharp
+
+    public async Task<GraphResponse> SendAsyncWithGraphResponse() {
+        using (var response = await this.SendRequestAsync(serializableObject, cancellationToken, completionOption).ConfigureAwait(false))
```

Comment:
I don't think we can have using here.  The response object would get disposed when we return the GraphResponse which holds a reference to the response object.

Reply:
That is very true. 🤦‍♂